src/glum/_glm.py

```diff
@@ -629,9 +629,16 @@ def is_pos_semidef(p: Union[sparse.spmatrix, np.ndarray]) -> Union[bool, np.bool
 
 def _group_sum(groups: np.ndarray, data: np.ndarray):
     """Sum over groups."""
-    out = np.empty((len(np.unique(groups)), data.shape[1]))
-    for i in range(data.shape[1]):
-        out[:, i] = np.bincount(groups, weights=data[:, i])
+    ngroups = len(np.unique(groups))
+    out = np.empty((ngroups, data.shape[1]))
+    if sparse.issparse(data) or isinstance(
+        data, (tm.SplitMatrix, tm.CategoricalMatrix)
+    ):
+        for i in range(data.shape[1]):
+            out[:, i] = (np.eye(ngroups)[:, groups] @ data.getcol(i)).ravel()
```

Comment:
Could we create `np.eye(ngroups)` before starting the `for` loop so we only need to create it once?

Reply:
I'm taking it that `np.bincount(groups, weights=data.getcol(i))` doesn't work?